engine/src/main/java/com/odysseusinc/arachne/executionengine/service/impl/DescriptorServiceImpl.java

```diff
@@ -154,12 +189,10 @@ private DescriptorBundle toBundle(Descriptor descriptor) {
         String bundleName = descriptor.getBundleName();
         String path = archiveFolder.map(folder -> folder.resolve(bundleName).toFile().getPath()).orElse(bundleName);
         File bundle = new File(path);
-        if (bundle.exists() && bundle.isFile()) {
-            return new DescriptorBundle(path, descriptor);
-        } else {
-            LOGGER.info("Descriptor [{}] matched, but bundle [{}] not found", descriptor.getLabel(), path);
-            return null;
+        if (!bundle.exists() || !bundle.isFile()) {
```

Comment:
What it the point to flip the condition? If descriptor bundle is not there, execution will not work with it.
This information should be returned to caller so it can react to it, not buried in log.

Reply:
Also, this change will in fact interfere with logging at line 132

Reply:
Execution will work with hosted R environment, this useful in development mode. By the way, how descriptor service should work in the non-isolated mode?

Reply:
"By the way, how descriptor service should work in the non-isolated mode?"
I have no idea. IMO it shouldn't be used at all for such a use case.
If caller is already aware about possible docker and tarball execution modes, it is only logical that it should be aware about hosted mode as well. However it was not in scope of the docker implementation.
AFAIK, we should treat hosted mode as legacy.